### PR TITLE
Changes needed in support of `alchemiscale-fah`

### PR DESCRIPTION
**Pull request by dotsdl**
This PR contains all changes needed for `alchemiscale-fah` MVP support.

### CODE REVIEW

**Comment by dotsdl** [comment]
@ijpulidos can I get an initial review on this one? I'm not sure we'll need any additional changes in `feflow` in support of https://github.com/openforcefield/alchemiscale-fah/pull/7, but I'll keep it in [WIP] state until I'm certain of that. A review will still be helpful to ensure there aren't any obvious gaps in the changes here.

**Comment by dotsdl** [comment]
Actually, I think @ijpulidos is on vacation starting today; can I get a review from you instead @IAlibay?

**Review by IAlibay — APPROVED** [review]

File: feflow/protocols/nonequilibrium_cycling.py [hunk 1]
```diff
@@ -325,9 +325,14 @@ def _execute(self, ctx, *, state_a, state_b, mapping, settings, **inputs):
             openmm.LocalEnergyMinimizer.minimize(context)
 
             # SERIALIZE SYSTEM, STATE, INTEGRATOR
+            thermodynamic_settings = settings.thermo_settings
+            temperature = to_openmm(thermodynamic_settings.temperature)
```
Comment (IAlibay) [inline feflow/protocols/nonequilibrium_cycling.py:329]: Possibly worth adding a comment on why this is necessary? Ideally a test too, but at least the former so we don't accidentally do a regression if we refactor stuff.

Comment (dotsdl) [reply feflow/protocols/nonequilibrium_cycling.py:332]: Done! Added comments indicating the importance of these changes to Folding@Home use!

**Comment by ijpulidos** [comment]
@dotsdl The CI failing is because we have __temporarily__ dropped support for GAFF, but we are still using it in our tests and these are not easy to adapt without some non-trivial amount of work. Thats also why it's working with older python (which install older versions of the dependencies and GAFF that work fine).

The changes in this PR are unrelated to that issue, this should be ready to be merged. LGTM!

**Pull request merged**